core/src/components/picker-column-internal/picker-column-internal.tsx

```diff
@@ -388,6 +389,21 @@ export class PickerColumnInternal implements ComponentInterface {
     });
   };
 
+  private focusActiveItem() {
+    const activeItem = this.activeItem;
+    if (activeItem) {
+      /**
+       * Focusing the active item on value change
+       * will announce the value to screen readers.
+       */
+      activeItem.focus();
+    }
+  }
+
+  private onFocus = () => {
+    this.focusActiveItem();
```

Comment:
Additionally we need to focus the active item manually when the picker column receives focus, so that the active item is correctly announced.

Reply:
Unable to make this improvement right now. The input mode implementation is not compatible with the button being focused when the column is focused. The input requires the keypress events to be bubbled from the picker column being focused.